src/AppBundle/Repository/PostRepository.php

```diff
@@ -38,4 +38,24 @@ public function findLatest($limit = Post::NUM_ITEMS)
             ->getResult()
         ;
     }
+
+    public function findByTerms(array $terms, $limit = Post::NUM_ITEMS)
+    {
+        $queryBuilder = $this->createQueryBuilder('p');
```

Comment:
You're right. I think now we should following styles already used in this project.